server/src/com/thoughtworks/go/server/service/GoConfigService.java

```diff
@@ -581,17 +585,33 @@ public void updateAgentAttributes(String uuid, String userName, String hostname,
             command.addCommand(GoConfigDao.updateApprovalStatus(uuid, false));
         }
 
-        if (enable.isFalse()){
+        if (enable.isFalse()) {
             command.addCommand(GoConfigDao.updateApprovalStatus(uuid, true));
         }
 
         if (hostname != null) {
             command.addCommand(new GoConfigDao.UpdateAgentHostname(uuid, hostname, userName));
         }
+
         if (resources != null) {
             command.addCommand(new GoConfigDao.UpdateResourcesCommand(uuid, new Resources(resources)));
         }
 
+        if (environments != null) {
+            Set<String> existingEnvironments = cruiseConfig().getEnvironments().environmentsForAgent(uuid);
```

Comment:
I think this should be a method, by itself, especially if we're thinking of an environments API.